src/control/joystick_manager.hpp

```diff
@@ -40,13 +47,15 @@ class JoystickManager final
 
   void bind_next_event_to(Control id);
 
-  void set_joy_controls(Control id, bool value);
+  void set_joy_controls(SDL_JoystickID joystick, Control id, bool value);
 
   void on_joystick_added(int joystick_index);
   void on_joystick_removed(int instance_id);
 
   int get_num_joysticks() const { return static_cast<int>(joysticks.size()); }
 
+  std::unordered_map<SDL_Joystick*, int>& get_joystick_mapping() { return joysticks; }
```

Comment:
Can this actually be marked const as well?

Reply:
`src/supertux/menu/multiplayer_player_menu` needs read-write access to the mapping (to avoid having multiple players bound to the same joystick)